### ISSUE TITLE
Improve C# -> F# lambda conversion codefix

### ISSUE THREAD

**Issue by psfinaki** [dotnet/fsharp#15499]
Currently codefix is activated for this code:

```fsharp
let incAll = List.map (n => n + 1)
```

But not for this code:
```fsharp
let getPositives = List.filter (n => n > 0)
```

Not a bug, but would be useful.

**Comment by nih0n** [comment]
Hey, done some research about this, the bug because of  the `>` character in the lambda body, for some reason it breaks something in the `TryRangeOfParenEnclosingOpEqualsGreaterUsage` during the `SyntaxTraversal.Traverse` process but I couldn't trace where exactly

**Comment by psfinaki** [comment]
Hi @nih0n! Thanks for taking a look into this. :) 

Indeed, it seems like a problem with the syntax traversal. Actually, I am quite sure the filtering we are doing in [this function](https://github.com/dotnet/fsharp/blob/256c7b240e063217a51b90d8886d0b789ceb066e/src/Compiler/Service/FSharpParseFileResults.fs#L310C1-L317C24) has a flaw. By then, we have identified that we're in the parenthesized statement and so we are looking what's inside. With `n => n + 1`, the parser thinks we are trying to somehow apply `n` to `n + 1` whereas with `n => n > 0`, it thinks we are trying to apply the whole `n => n` to `0` (basically it thinks we are comparing the former to the latter hence the operator in question is `>` not `=>`). 

In syntax traversal terms, we are dealing with `SynExpr.App`. It has a part containing a functor `funcExpr` and a part containing an argument `argExpr`. And here, most probably in the code above, we should actually look at `argExpr` instead of `funcExpr` (and then within `argExpr` probably to the `funcExpr` again). Since we want both things working, it will probably mean just adding another pattern matching case to the expression.

Does it make sense? Do you want to give it a stab? At least this should be quite easy to test and debug now - the latest main contains `ConvertCSharpLambdaToFSharpLambdaTests` for unit testing of this stuff.

**Comment by brianrourkeboll** [comment]
> With `n => n + 1`, the parser thinks we are trying to somehow apply `n` to `n + 1` whereas with `n => n > 0`, it thinks we are trying to apply the whole `n => n` to `0` (basically it thinks we are comparing the former to the latter hence the operator in question is `>` not `=>`).

It's because `=>` and `>` have the same precedence, while `+` has higher precedence than `=>` (and all are left-associative). So `n => n > 0` and `n => n + 1` will have [different ASTs](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQ2wAuYA%2BmAC8YZMIB8osJIAMAHQBOc5Nz6CRYoZLEBqMAEY8BSLAAeAByrJoFHgoCusEAF8gA).

**Comment by psfinaki** [comment]
Yeah. I've learned something about op precedence since then as well 👀
